### PR TITLE
Feature/save manager

### PR DESCRIPTION
**Pull request by JessieVela**
Created a base save provider where it gives the option to save in SQLite database. Currently, Player is implemented and created so it is able to be saved to the database. Removed the .idea folder from the project.

### CODE REVIEW

File: src/main/java/xyz/devosmium/games/textadventureengine/mobiles/Mobile.java [hunk 1]
```diff
@@ -1,86 +1,95 @@
...
+ * interact with the world around them. Mobiles are assigned a unique identifier,
+ * so that they can be saved in the database.
  */
 public class Mobile {
 
-  // Declare all variables
-  private int maxHealth;
-  private int health;
+    // Declare all variables
+    private String playerID;
```
Comment (Muirrum) [inline src/main/java/xyz/devosmium/games/textadventureengine/mobiles/Mobile.java]: I would move this to the Player class

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        statement = connection.createStatement();
+        statement.setQueryTimeout(30);
+
+        String sql = "CREATE TABLE IF NOT EXISTS PLAYER " +
+                "(PLAYER_ID      TEXT PRIMARY KEY  NOT NULL, " +
+                " NAME           TEXT    NOT NULL, " +
+                " MAX_HEALTH     INT     NOT NULL, " +
+                " HEALTH         INT     NOT NULL, " +
+                " LEVEL          INT     NOT NULL, " +
+                " GOLD           INT     NOT NULL )";
```
Comment (Muirrum) [inline src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:67]: Just making a note to myself that it needs to save inventories as well, though how to do that?

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    private Statement statement;
+    private Player player;
+
+    /**
+     * SQLite will instantiate a database called game.db. This database will hold all object information of the world
+     * to be stored in the database.
+     *
+     * @param player player will allow access to methods used for retrieving stats for saving
+     */
+    public void SQLiteSave(Player player) {
```
Comment (Muirrum) [inline src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:37]: Saving looks mostly good, what about loading?

**Review by Muirrum — CHANGES_REQUESTED** [review]
Saving here looks good, but it needs to include loading.

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java (hunk 3 above)
Comment (JessieVela) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:37]: Slipped my mind about loading. Will add a loading method.

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java (hunk 2 above)
Comment (JessieVela) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:67]: I would do the same as stats. Make a table and have a foreign key reference the player.

File: src/main/java/xyz/devosmium/games/textadventureengine/mobiles/Mobile.java (hunk 1 above)
Comment (JessieVela) [reply src/main/java/xyz/devosmium/games/textadventureengine/mobiles/Mobile.java]: Done.

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java (hunk 2 above)
Comment (Muirrum) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:67]: But then how would it scale to different inventory sizes?

Comment (JessieVela) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:67]: Hmmm well you could insert into the tables the different inventories and update their values as a player gets them.
PlayerID | Potion | Key | Revive | Elixer | Stamina |
123s      |     2      |   0   |     1       |   0     |      4        |

Then you would query the table for items greater than 0. 

If more items get added to the game you would have to go into the code and add them to the table. This is what comes to mind and im sure there are better ways to achieve this.

Comment (Muirrum) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:67]: Could also have just an inventory table, which had a list of items and a reference back to the player.

**Comment by JessieVela** [comment]
Quick update: Still working on the Loading portion. Its more complicated than I anticipated.

**Comment by Muirrum** [comment]
No problem, thanks for the update




---- On Wed, 30 Oct 2019 14:51:54 -0400 JessieVela <mailto:notifications@github.com> wrote ----


Quick update: Still working on the Loading portion. Its more complicated than I anticipated.

File: src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java (hunk 3 above)
Comment (JessieVela) [reply src/main/java/xyz/devosmium/games/textadventureengine/save/SaveProvider.java:37]: Finally got it finished.

**Comment by JessieVela** [comment]
Finally, finished doing all the changes. #5

**Review by Muirrum — CHANGES_REQUESTED** [review]
If you could merge the commit where I fixed the gradle wrapper, that would do it! (commit b2c94a3cf) this should be able to be accomplished by merging develop into the branch.

**Comment by JessieVela** [comment]
Done.

**Review by Muirrum — COMMENTED** [review]
Build [fails](https://scans.gradle.com/s/z5hxcxthblol4)

**Comment by JessieVela** [comment]
Finally have time to take a look. Its failing because of the command interface. This made it into develop. I can fix it and push the changes.
Screen Shot 2019-11-16 at 9 09 48 AM

**Comment by JessieVela** [comment]
On further review this is going to take a minute to fix. My branch didn't create this and might not have time to solve it but I will try.

**Comment by JessieVela** [comment]
The whole command engine has issues. It's not even complete.

**Pull request closed without merging**